src/lib.rs

```diff
@@ -1382,14 +1380,13 @@ impl<'a> Drop for ComputePass<'a> {
 
 impl Queue {
     /// Submits a series of finished command buffers for execution.
-    pub fn submit(&mut self, command_buffers: &[CommandBuffer]) {
-        self.temp_command_buffers.clear();
-        self.temp_command_buffers
-            .extend(command_buffers.iter().map(|cb| cb.id));
+    pub fn submit(&self, command_buffers: &[CommandBuffer]) {
```

Comment:
upon some thinking, I think this is safe to do. We just need to make sure the expectations, especially with regards to threading, are well defined by wgpu-native and wgpu-core

Reply:
It seems to me most functions that are currently `&mut seulf` could be safely changed to `&self` right now, because wgpu-core will acquire locks anyways.

Reply:
`&mut self` also makes sure the access is synchronized, i.e. you don't want the clients to concurrently record to the same pass, for example. This is directly related to the non-send non-sync enforcement that was put in for the command encoders.
Even if we can, I don't think anything with regards to command recording should be changed to `&self`.

Reply:
That makes sense yea